proposals/lip-Introduce_cross_chain_messages.md

```diff
@@ -0,0 +1,650 @@
+         CCM is discarded and has no further effect
+      elif moduleID is not supported:
+         newCCM = CCM
+         newCCM.fee = 0
+         newCCM.status = CCM_STATUS_MODULE_NOT_SUPPORTED
+         swap newCCM.receivingChainID and newCCM.sendingChainID
+         process(newCCM) 
+      elif crossChainCommandID is not supported:
+         newCCM = CCM
+         newCCM.fee = 0
+         newCCM.status = CCM_STATUS_CROSS_CHAIN_COMMAND_NOT_SUPPORTED 
+         swap newCCM.receivingChainID and newCCM.sendingChainID
+         process(newCCM) 
+```
+
+```python
+tryToForward(CCM):
+   let partnerChainID = getPartnerChainID(CCM.receivingChainID)
+
+   if account(partnerChainID).status == CHAIN_ACTIVE and isLive(partnerChainID):
+      addToOutbox(partnerChainID, CCM)
+   elif CCM.status == CCM_STATUS_OK:
+      newCCM = CCM
+      newCCM.fee = 0
+      newCCM.status = CCM_STATUS_CHANNEL_UNAVAILABLE  
+      swap newCCM.receivingChainID and newCCM.sendingChainID
+      process(newCCM)
+      
+      STMParams = {chainID: CCM.receivingChainID,
+                  stateRoot: account(CCM.receivingChainID).lastCertifiedStateRoot}
+                  serialized using sidechainTerminatedCCMParamsSchema
+      STM = createCrossChainMessage(MODULE_ID_INTEROPERABILITY, 
+                                    CROSS_CHAIN_COMMAND_ID_SIDECHAIN_TERMINATED,
+                                    CCM.sendingChainID,
+                                    0,
+                                    STMParams)
+      addToOutbox(STM)
+   else:
+      CCM is discarded and has no further effect
+```
+The `addToOutbox`, `getPartnerChainID` and `isLive` functions are defined in [LIP "Introduce Interoperability module"][base-interoperability-LIP].
+
+
+### Cross-chain Commands
+
+
+#### Cross-chain Update Receipt
+
+The cross-chain update receipt (CCU receipt) is a CCM with 
+
+*   `moduleID = MODULE_ID_INTEROPERABILITY`,
+*   `crossChainCommandID  = CROSS_CHAIN_COMMAND_ID_CCU_RECEIPT`.
+
+The main role of the cross-chain update receipt is to inform chains that a cross-chain update transaction has been posted on the partner chain.
+
+##### Parameters
+
+The `params` schema for CCU receipts is:
+
+```java
+ccuReceiptParamsSchema = {
+    "type": "object",
+    "required": [
+        "paidFee", 
+        "relayerPublicKey", 
+        "partnerChainInboxSize"
+    ],
+    "properties": {
+        "paidFee": {
+            "dataType": "uint64",
+            "fieldNumber": 1 
+        },
+        "relayerAddress": {
+            "dataType": "bytes",
+            "fieldNumber": 2 
+        },
+        "partnerChainInboxSize": {
+            "dataType": "uint64",
+            "fieldNumber": 3
+        }
+    }
+}
+```
+*   `paidFee`: fee paid by the relayer for the cross-chain update transaction,
+*   `relayerAddress`: address of the relayer posting the cross-chain update transaction. This property must be 20 bytes long.
```

Comment:
```suggestion
*   `relayerAddress`: address of the relayer posting the cross-chain update transaction. This property must be 20 bytes long,
```

Reply:
or end with a period the previous bullet point, for consistency

Reply:
ja, ended the previous one with a bullet.